src/ResourceManagement/Network/AzureFirewallImpl.cs

```diff
@@ -457,37 +462,17 @@ protected async override Task<AzureFirewallInner> GetInnerAsync(CancellationToke
 
         AzureFirewall.Definition.IWithNatRuleCollectionSettings IInDefinition<AzureFirewall.Definition.IWithNatRuleCollectionSettings>.Attach()
         {
-            return AttachNatRule();
+            return Attach();
         }
 
         AzureFirewall.Definition.IWithCreate IInDefinition<AzureFirewall.Definition.IWithCreate>.Attach()
         {
-            if (applicationRuleCollection != null)
-            {
-                Inner.ApplicationRuleCollections.Add(applicationRuleCollection);
-                applicationRuleCollection = null;
-            }
-            else if (natRuleCollection != null)
-            {
-                Inner.NatRuleCollections.Add(natRuleCollection);
-                natRuleCollection = null;
-            }
-            else if (networkRuleCollection != null)
-            {
-                Inner.NetworkRuleCollections.Add(networkRuleCollection);
-                networkRuleCollection = null;
-            }
-            else if (ipConfiguration != null)
-            {
-                Inner.IpConfigurations.Add(ipConfiguration);
-                ipConfiguration = null;
-            }
-            return this;
+            return Attach();
```

Comment:
what's impact of remove existing appending rule action?

Reply:
I just move this appending action into define method. This is to align with update operations which don't required appending action.